scality_sproxyd_client/sproxyd_client.py

```diff
@@ -158,11 +187,21 @@ def get_url_for_object(self, name):
         path. The object `name` is properly escaped by `urllib.quote` which
         preserves '/' by default.
         '''
+
         try:
-            sproxyd_url = self.sproxyd_urls.next()
+            endpoint = self.get_next_endpoint()
         except StopIteration:
             raise exceptions.SproxydException("No Sproxyd endpoint alive")
-        return sproxyd_url + urllib.quote(name)
+
+        path = endpoint.path.strip('/')
```

Comment:
This is a bit suboptimal considering we might get/put 3k objects per second. Though I am okay with it because I bet `strip` is cheap.